archunit/src/main/java/com/tngtech/archunit/core/importer/DependencyResolutionProcess.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2014-2022 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.core.importer;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+
+import com.tngtech.archunit.ArchConfiguration;
+import com.tngtech.archunit.core.importer.ImportedClasses.ImportedClassState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.tngtech.archunit.core.importer.ImportedClasses.ImportedClassState.HAD_TO_BE_IMPORTED;
+import static java.lang.System.lineSeparator;
+
+class DependencyResolutionProcess {
+    private static final Logger log = LoggerFactory.getLogger(DependencyResolutionProcess.class);
+
+    static final String DEPENDENCY_RESOLUTION_PROCESS_PROPERTY_PREFIX = "import.dependencyResolutionProcess";
+
+    static final String MAX_ITERATIONS_FOR_MEMBER_TYPES_PROPERTY_NAME = "maxIterationsForMemberTypes";
+    static final String MAX_ITERATIONS_FOR_ACCESSES_TO_TYPES_PROPERTY_NAME = "maxIterationsForAccessesToTypes";
+    static final String MAX_ITERATIONS_FOR_SUPERTYPES_PROPERTY_NAME = "maxIterationsForSupertypes";
+    static final String MAX_ITERATIONS_FOR_ENCLOSING_TYPES_PROPERTY_NAME = "maxIterationsForEnclosingTypes";
+    static final String MAX_ITERATIONS_FOR_ANNOTATION_TYPES_PROPERTY_NAME = "maxIterationsForAnnotationTypes";
+    static final String MAX_ITERATIONS_FOR_GENERIC_SIGNATURE_TYPES_PROPERTY_NAME = "maxIterationsForGenericSignatureTypes";
+
+    static final int MAX_ITERATIONS_FOR_MEMBER_TYPES_DEFAULT_VALUE = 1;
+    static final int MAX_ITERATIONS_FOR_ACCESSES_TO_TYPES_DEFAULT_VALUE = 1;
+    static final int MAX_ITERATIONS_FOR_SUPERTYPES_DEFAULT_VALUE = -1;
+    static final int MAX_ITERATIONS_FOR_ENCLOSING_TYPES_DEFAULT_VALUE = -1;
+    static final int MAX_ITERATIONS_FOR_ANNOTATION_TYPES_DEFAULT_VALUE = -1;
+    static final int MAX_ITERATIONS_FOR_GENERIC_SIGNATURE_TYPES_DEFAULT_VALUE = -1;
+
+    private final ResolutionRun resolutionRun = new ResolutionRun();
```

Comment:
As there's a 1:1 correspondence between `DependencyResolutionProcess` and `ResoulutionRun` (even though `resolve` may trigger multiple "runs"):
Is it really helpful to have `ResoulutionRun` as a separate concept, with all the delegation boilerplate code?

If you merged those two classes into one, the fields for the actually configured values (such as `maxRunsForMemberTypes`, etc.) could move next to the constants (`MAX_ITERATIONS_FOR_MEMBER_TYPES_PROPERTY_NAME` etc. and `MAX_ITERATIONS_FOR_MEMBER_TYPES_DEFAULT_VALUE` etc.), which would IMO be nice as well.

Reply:
Yeah, maybe right :thinking: Originally I thought there would be more left in the outer class, but turned out that pretty much all the logic ended inside of `ResolutionRun`. I guess a `runNumber` in `DependencyResolutionProcess` will be good enough to get it. Will be super painful to fix this up in the history though :rofl:

Reply:
Well, I wouldn't insist on it, but I know you like this kind of ~pain~ clean history... Oh, I know you do so much! 😝

Reply:
Already done :wink: Like it never was over-engineered :wink:

Reply:
+🙈